### ISSUE TITLE
TranslitUrl class doesn't check the current locale

### ISSUE THREAD

**Issue by jasperzeinstra** [integer-net/magento2-german-umlaut-urls#2]
Cool module, but the TranslitUrl class doesn't check the locale. For other languages then DE you want "ö" to be transformed to "o"

**Comment by avstudnitz** [comment]
True, the module doesn't support that (yet). I could think of a configuration switch where you could turn on or off the feature on a store view base - on the other hand, I am not sure if the frontend store is even set when generating URLs, so it could be difficult to retrieve the context.

**Comment by jasperzeinstra** [comment]
I think it's best to get the current store. Then check that locale. If it's Geman then replace/add the $convertTable. So adding something to the _construct function to check the current scope.

```
public function __construct(
    \Magento\Framework\App\Config\ScopeConfigInterface $config
    \Magento\Store\Model\StoreManagerInterface $storeManager
)
{
    $this->storeManager = $storeManager;
    parent::__construct($config);
}
```

When I have time I will check if this works.

**Comment by Crissov** [comment]
I wanted to leave a brief comment about CLDR/ICU data, but this quickly became a rabbit hole I went down, and in the end I wrote a bug report to Magento instead: https://github.com/magento/magento2/issues/37802
